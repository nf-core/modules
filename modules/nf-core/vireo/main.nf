process VIREO {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/vireosnp:0.5.8--pyh7cba7a3_0' :
        'biocontainers/vireosnp:0.5.8--pyh7cba7a3_0' }"

    input:
    tuple val(meta), path(cell_data), val(n_donor), path(donor_file), path(vartrix_data)
    output:
    tuple val(meta), path('*_summary.tsv')        , emit: summary
    tuple val(meta), path('*_donor_ids.tsv')      , emit: donor_ids
    tuple val(meta), path('*_prob_singlet.tsv.gz'), emit: prob_singlets
    tuple val(meta), path('*_prob_doublet.tsv.gz'), emit: prob_doublets
    path 'versions.yml'                           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args   = task.ext.args   ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def input  = cell_data       ? "-c ${cell_data}" : "--vartrixData ${vartrix_data}"

    """
    vireo \\
        $input \\
        -N ${n_donor} \\
        -d ${donor_file} \\
        -p $task.cpus \\
        -o . \\
        $args

    mv summary.tsv ${prefix}_summary.tsv
    mv donor_ids.tsv ${prefix}_donor_ids.tsv
    mv prob_singlet.tsv.gz ${prefix}_prob_singlet.tsv.gz
    mv prob_doublet.tsv.gz ${prefix}_prob_doublet.tsv.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vireo: \$(vireo | sed '1!d ; s/Welcome to vireoSNP //; s/!//')
    END_VERSIONS
    """

    stub:
    def args   = task.ext.args   ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}_summary.tsv
    touch ${prefix}_donor_ids.tsv
    echo "" | gzip > ${prefix}_prob_singlet.tsv.gz
    echo "" | gzip > ${prefix}_prob_doublet.tsv.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vireo: \$(vireo | sed '1!d ; s/Welcome to vireoSNP //; s/!//')
    END_VERSIONS
    """
}
