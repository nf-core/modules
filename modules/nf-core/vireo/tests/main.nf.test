// nf-core modules test vireo
nextflow_process {

    name "Test Process VIREO"
    script "../main.nf"
    process "VIREO"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "vireo"
    tag "samtools/index"
    tag "cellsnp/modea"

    test("demultiplexing") {
        setup {
            run("SAMTOOLS_INDEX") {
                script "../../samtools/index/main.nf"
                process {
                    """
                    input[0] = [
                    [ id:'sample1' ],
                    file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/chr21.bam', checkIfExists: true) ]
                    """
                }
            } 
            run("CELLSNP_MODEA") {
                script "../../cellsnp/modea/main.nf"
                process {
                    """
                    input[0] = SAMTOOLS_INDEX.out.bai.collect{ meta, bai -> bai }.map{ 
                               bai -> [[id: 'sample1'], 
                                        file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/chr21.bam', checkIfExists: true),
                                        bai,
                                        file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/donor_genotype_chr21.vcf', checkIfExists: true),
                                        file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/barcodes.tsv', checkIfExists: true)]}
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = CELLSNP_MODEA.out.cell.collect{ meta, cell -> cell }.map{ 
                            cell -> [[id:'test'],
                                      cell,
                                      2,
                                      file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/donor_genotype_chr21.vcf', checkIfExists: true),
                                      []
                                    ]}
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("demultiplexing - stub") {
        options "-stub"

        when {
            process {
                """
                input[0] = [[id:'sample1'],
                            [],
                            2,
                            file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/donor_genotype_chr21.vcf', checkIfExists: true),
                            []
                            ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
