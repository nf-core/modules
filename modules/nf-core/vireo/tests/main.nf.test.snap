{
    "demultiplexing - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "sample1"
                        },
                        "sample1_summary.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "sample1"
                        },
                        "sample1_donor_ids.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "sample1"
                        },
                        "sample1_prob_singlet.tsv.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "sample1"
                        },
                        "sample1_prob_doublet.tsv.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "4": [
                    "versions.yml:md5,b16f367dd80f537073c53ccaf9fd547d"
                ],
                "donor_ids": [
                    [
                        {
                            "id": "sample1"
                        },
                        "sample1_donor_ids.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "prob_doublets": [
                    [
                        {
                            "id": "sample1"
                        },
                        "sample1_prob_doublet.tsv.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "prob_singlets": [
                    [
                        {
                            "id": "sample1"
                        },
                        "sample1_prob_singlet.tsv.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "sample1"
                        },
                        "sample1_summary.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b16f367dd80f537073c53ccaf9fd547d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T22:38:03.769285328"
    },
    "demultiplexing": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_summary.tsv:md5,94b0ce40cc86bbe7fbacebf98e685ee7"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_donor_ids.tsv:md5,2d3eadd8d1a3f26428b383e7a98d2ced"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test_prob_singlet.tsv.gz:md5,d1e6d055fda3c96713912da79994b48c"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test_prob_doublet.tsv.gz:md5,6d7e979931de2b66b89618be47fc079c"
                    ]
                ],
                "4": [
                    "versions.yml:md5,b16f367dd80f537073c53ccaf9fd547d"
                ],
                "donor_ids": [
                    [
                        {
                            "id": "test"
                        },
                        "test_donor_ids.tsv:md5,2d3eadd8d1a3f26428b383e7a98d2ced"
                    ]
                ],
                "prob_doublets": [
                    [
                        {
                            "id": "test"
                        },
                        "test_prob_doublet.tsv.gz:md5,6d7e979931de2b66b89618be47fc079c"
                    ]
                ],
                "prob_singlets": [
                    [
                        {
                            "id": "test"
                        },
                        "test_prob_singlet.tsv.gz:md5,d1e6d055fda3c96713912da79994b48c"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test"
                        },
                        "test_summary.tsv:md5,94b0ce40cc86bbe7fbacebf98e685ee7"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b16f367dd80f537073c53ccaf9fd547d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T22:35:04.435257747"
    }
}